bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/service/AbstractWatchService.java

```diff
@@ -77,7 +85,9 @@ protected void initializeWatchService() {
                 if (watchSubDirectories()) {
                     watchService = FileSystems.getDefault().newWatchService();
 
-                    Files.walkFileTree(toWatch, new SimpleFileVisitor<Path>() {
+                    // walk through all folders and follow symlinks
+                    Files.walkFileTree(toWatch, Sets.newHashSet(FileVisitOption.FOLLOW_LINKS), MAX_DEPTH, 
```

Comment:
> How is the max depth related to "adding follow symlink option"?

IMHO it's not related, but it's a required parameter when using options: 
https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#walkFileTree-java.nio.file.Path-java.util.Set-int-java.nio.file.FileVisitor-

>Why do we need to restrict the depth of the abstract watch service? Isn't it a specific usage if the depth is limited to a specific value? 

O.K. good point. I thought on my case with only depth of 2 levels (conf/ and conf/rules, etc.). Maybe we should use `Integer.MAX_VALUE` so all levels should be visited. Preventing cycles is done by the Files implementations. 

Do you prefer using `Integer.MAX_VALUE `